File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/LayoutSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.PopupPosition
import com.usercentrics.sdk.UsercentricsLayout

internal fun Any?.deserializeLayout(): UsercentricsLayout? {
    this ?: return null

    return when (this as String) {
        "FULL" -> UsercentricsLayout.Full
        "SHEET" -> UsercentricsLayout.Sheet
        "POPUP_BOTTOM" -> UsercentricsLayout.Popup(PopupPosition.BOTTOM)
        "POPUP_CENTER" -> UsercentricsLayout.Popup(PopupPosition.CENTER)
        else -> throw IllegalArgumentException("Invalid layout: $this")
    }
}

File: ios/Classes/Serializer/LayoutSerializer.swift
import UsercentricsUI

extension UsercentricsLayout {
    static func initialize(from value: Any?) -> UsercentricsLayout? {
        let valueString = value as! String
        switch valueString {
        case "SHEET":
            return .sheet
        case "FULL":
            return .full
        case "POPUP_CENTER":
            return .popup(position: .center)
        case "POPUP_BOTTOM":
            return .popup(position: .bottom)
        default:
            assert(false)
            return nil
        }
    }
}
